extension.js

```diff
@@ -36,6 +36,10 @@ class ExtensionDBusInterface {
             return;
 
         Main.wm.skipNextEffect(current_window.get_compositor_private());
+        if (settings.get_boolean('window-maximize')) {
+            settings.set_boolean('window-maximize', false);
+            settings.set_double('window-height', 1.0);
+        }
```

Comment:
This last commit seems to work well with all "constraints" but I'm wondering whether it would more sense to call here `update_window_height` like:
```suggestion
     update_window_height()
```
so that the same code is not repeated in several places. The problem is I don't know whether there are some conflicts between what `update_window_geometry` and `BeginResize` do.

Reply:
See the comments above